iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +28,649 @@ using namespace iox::capro;
+    auto serialObj = iox::cxx::Serialization::create(testService,
+                                                     testInstance,
+                                                     testEvent,
+                                                     testServiceID,
+                                                     testInstanceID,
+                                                     testEventID,
+                                                     testHash[0],
+                                                     testHash[1],
+                                                     testHash[2],
+                                                     testHash[3],
+                                                     true);
+
+    ServiceDescription serviceDescription1 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+    ServiceDescription serviceDescription2 = ServiceDescription(serialObj);
+
+    EXPECT_FALSE(serviceDescription1 == serviceDescription2);
+}
+
+TEST_F(ServiceDescription_test, TwoServiceDescriptionsWithDifferentEventStringsAreNotEqual)
+{
+    uint16_t testServiceID = 1U;
+    uint16_t testEventID = 2U;
+    uint16_t testInstanceID = 3U;
+    IdString_t testService = "1";
+    IdString_t testEvent = "event";
+    IdString_t testInstance = "3";
+    ServiceDescription::ClassHash testHash = {1, 2, 3, 4};
+    auto serialObj = iox::cxx::Serialization::create(testService,
+                                                     testInstance,
+                                                     testEvent,
+                                                     testServiceID,
+                                                     testInstanceID,
+                                                     testEventID,
+                                                     testHash[0],
+                                                     testHash[1],
+                                                     testHash[2],
+                                                     testHash[3],
+                                                     true);
+
+    ServiceDescription serviceDescription1 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+    ServiceDescription serviceDescription2 = ServiceDescription(serialObj);
+
+    EXPECT_FALSE(serviceDescription1 == serviceDescription2);
+}
+
+TEST_F(ServiceDescription_test, TwoServiceDescriptionsWithDifferentValidIDsAreNotEqual)
+{
+    uint16_t testServiceID = 1U;
+    uint16_t testEventID = 2U;
+    uint16_t testInstanceID = 3U;
+    ServiceDescription serviceDescription1 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+    ServiceDescription serviceDescription2 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+
+    EXPECT_FALSE(serviceDescription1 != serviceDescription2);
+}
+
+TEST_F(ServiceDescription_test, ServiceDescriptionClassHashCtorCreatesClassHashWithValuesPassedToTheCtor)
```

Comment:
There are some test cases for the ClassHash missing. For example the `[]` operator or the initialization of it.